elements/pf-select/demo/option-variations.html

```diff
@@ -0,0 +1,20 @@
+<link rel="stylesheet" href="demo.css">
+<script type="module" src="pf-select.js"></script>
+<section>
+  <h2>Option variations</h2>
+  <p>
+    Below are the option variants:
+  </p>
+  <pf-select always-open>
+    <pf-select-option value="Basic">Basic option</pf-select-option>
```

Comment:
is `pf-option` better? like, would it work with other kinds of listbox, autocomplete, etc?

Reply:
If we want to match upstream Patternfly, the components would be:
- `pf-select` for PatternFly's `Select`
- `pf-select-list` for PatternFly's `SelectList`
- `pf-select-group` for PatternFly's `SelectGroup`
- `pf-select-option` for PatternFly's `SelectOption`

`combobox` / `autocomplete` is actually the `typeahead` feature in PatternFly's `Select`

Basically this one component does all things that are for accessibility purposes considered listboxes, with the exception of if we ever implement `DualListSelector`, but it uses `DualListSelectorPane`, `DualListSelectorList`, and `DualListSelectorListItem `.

Reply:
we don't need to match pf upstreams apis, only their end-user experience